jwala-services/src/main/java/com/cerner/jwala/service/jvm/state/JvmStateReceiverAdapter.java

```diff
@@ -69,18 +70,73 @@ public void receive(final Message jGroupMsg) {
         }
     }
 
+    private JvmState getJvmState(final Map serverInfoMap) {
+
+        // first check if the key types are string
+        final Set keys = serverInfoMap.keySet();
+        if (keys.isEmpty()) {
+            return null;
+        }
+
+        final Object initialKey = keys.iterator().next();
+        if (initialKey instanceof String) {
```

Comment:
We can't check for type of ReportingJmsMessageKey either. The main problem is that the old way sends the message map with the ReportingJmsMessageKey as a serialized object, but the ReportingJmsMessageKey only exists in the ctp-infrastructure libraries. So, jwala has no idea what this class is and we can't import it without adding a dependency to ctp-infrastructure which we definitely don't want.

It's unfortunate but we can only rely on the keys being Strings to determine which version of the message we're receiving. We really can't even do keys.contains(STATE_KEY) to check for a ReportingJmsMessageKey because we can't create a ReportingJmsMessageKey object.